### PR TITLE
feat: DTOSS-5711 lets encrypt certificate automation

### PR DESCRIPTION
**Pull request by patrickmoore-nc**
## Description

- Invoke new Terraform module for requesting SSL certificates, renewing them, and storing them in Key Vault.
- Add Key Vault to Hub for certificate storage, using private endpoint.
- Pipeline must now use build agent from Managed DevOps Pool to access this Key Vault.

## Context

Let's Encrypt offers free browser-trusted SSL certificates with a validity of 3 months, provided DNS domain ownership can be authenticated. By automating the requesting of these certificates using the certbot tool with an Azure DNS authoriser plugin, and persisting the certbot state in a Storage Account the entire process can be wrapped in Terraform.

When the pipeline runs, any existing certificate with less than 30 days until expiry will be automatically renewed, stored in the specified Key Vaults, and resources which use this certificate will be updated.

## Testing
- [Successful build of Dev Hub](https://dev.azure.com/nhse-dtos/dtos-hub/_build/results?buildId=6230&view=results)

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

**Review by josielsouzanordcloud — APPROVED** [review]

File: infrastructure/baseline.tf [hunk 1]
```diff
@@ -15,10 +15,7 @@ resource "azurerm_resource_group" "rg_project" {
 }
 
 locals {
-  # Get the primary region from the regions map:
-  primary_region = {
-    for region_key, region in var.regions :
-  region_key => region if region.is_primary_region }
+  primary_region = [for k,v in var.regions : k if v.is_primary_region][0]
```
Comment (mrlockstar) [inline infrastructure/baseline.tf:18]: I suspect if you did terraform fmt that the space between k,v would be added. So this suggests to me that you have not done terraform fmt. Can you run it please?

File: infrastructure/environments/development.tfvars [hunk 2]
```diff
@@ -126,6 +130,15 @@ avd_vm_count          = 1
...
 avd_admins_group_name = "DToS-hub-dev-uks-hub-virtual-desktop-User-ADMIN-Login"
 
+dns_zone_name                = "dev.nationalscreening.nhs.uk"
+dns_zone_resource_group_name = "rg-hub-dev-uks-public-dns-zones"
+
+lets_encrypt_certificates = {
+  wildcard         = "*.dev.nationalscreening.nhs.uk"
+  wildcard_private = "*.private.dev.nationalscreening.nhs.uk"
+}
+lets_encrypt_contact_email = "josiel.souza1@nhs.net"
```
Comment (mrlockstar) [inline infrastructure/environments/development.tfvars:140]: You really want to put a personal email in here? Should it not be a shared mailbox as Josiel might roll off the project in the future he will not have access to that mailbox.

**Review by mrlockstar — COMMENTED** [review]
very minor things to change.

**Review by josielsouzanordcloud — APPROVED** [review]

File: infrastructure/baseline.tf (hunk 1 above)
Comment (patrickmoore-nc) [reply infrastructure/baseline.tf]: Done

File: infrastructure/environments/development.tfvars (hunk 2 above)
Comment (patrickmoore-nc) [reply infrastructure/environments/development.tfvars]: Moved to variable group

**Review by mrlockstar — APPROVED** [review]

**Pull request merged**